### PR TITLE
Adding framework for multinode tests

### PR DESCRIPTION
**Pull request by Ajay191191**

### CODE REVIEW

File: rl_coach/presets/Mujoco_ClippedPPO.py [hunk 1]
```diff
@@ -56,6 +56,8 @@
 # Environment #
 ###############
 env_params = GymVectorEnvironment(level=SingleLevelSelection(mujoco_v2))
+# Set the target success
+env_params.target_success_rate = 1.0
```
Comment (balajismaniam) [inline rl_coach/presets/Mujoco_ClippedPPO.py]: This should be fine as it is only used when `-asc` flag is passed to Coach.

File: rl_coach/tests/test_dist_coach.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+def test_dc(command, image, memory_backend, s3_end_point, s3_bucket_name, s3_creds_file, config_file):
+    clear_bucket(s3_end_point, s3_bucket_name)
+    command = command.format(template=config_file).split(' ')
+    test_command(command)
+
+
+def get_tests():
+    tests = [
+        'rl_coach/coach.py -p CartPole_ClippedPPO -dc -e sample -dcp {template} --dump_worker_logs --checkpoint_save_secs 20 -asc --test_preset',
```
Comment (balajismaniam) [inline rl_coach/tests/test_dist_coach.py]: --checkpoint_save_secs can be removed in both these commands.

File: rl_coach/coach.py [hunk 3]
```diff
@@ -569,6 +572,9 @@ def get_argument_parser(self) -> argparse.ArgumentParser:
         parser.add_argument('--dump_worker_logs',
                             help="(flag) Only used in distributed coach. If set, the worker logs are saved in the experiment dir",
                             action='store_true')
+        parser.add_argument('--test_preset',
```
Comment (balajismaniam) [inline rl_coach/coach.py]: s/--test_preset/--multi_node_test?

File: rl_coach/training_worker.py [hunk 4]
```diff
@@ -29,7 +30,7 @@ def data_store_ckpt_save(data_store):
         time.sleep(10)
 
 
-def training_worker(graph_manager, task_parameters):
+def training_worker(graph_manager, task_parameters, test_preset):
```
Comment (balajismaniam) [inline rl_coach/training_worker.py]: s/test_preset/multi_node_test

File: rl_coach/training_worker.py [hunk 5]
```diff
@@ -50,6 +51,11 @@ def training_worker(graph_manager, task_parameters):
     while steps < graph_manager.improve_steps.num_steps:
 
         graph_manager.phase = core_types.RunPhase.TRAIN
+        if test_preset and graph_manager.get_current_episodes_count() > graph_manager.preset_validation_params.max_episodes_to_achieve_reward:
```
Comment (balajismaniam) [inline rl_coach/training_worker.py]: s/test_preset/multi_node_test

File: rl_coach/coach.py (hunk 3 above)
Comment (balajismaniam) [reply rl_coach/coach.py]: @galleibo-intel I'm hoping this is fine with you?

**Comment by balajismaniam** [comment]
@galleibo-intel Do you have any comments on this? If not, I would like to merge this.

File: rl_coach/graph_managers/graph_manager.py [hunk 6]
```diff
@@ -706,3 +701,13 @@ def close(self) -> None:
         """
         for env in self.environments:
             env.close()
+
+    def get_current_episodes_count(self):
+        return self.current_step_counter[EnvironmentEpisodes]
```
Comment (galleibo-intel) [inline rl_coach/graph_managers/graph_manager.py]: Please add missing docstring.

File: rl_coach/graph_managers/graph_manager.py [hunk 7]
```diff
@@ -706,3 +701,13 @@ def close(self) -> None:
...
+
+    def flush_finished(self):
+        if self.task_parameters.checkpoint_save_dir and os.path.exists(self.task_parameters.checkpoint_save_dir):
```
Comment (galleibo-intel) [inline rl_coach/graph_managers/graph_manager.py]: Please add missing docstring.

File: rl_coach/tests/test_dist_coach.py [hunk 8]
```diff
@@ -0,0 +1,101 @@
+
```
Comment (galleibo-intel) [inline rl_coach/tests/test_dist_coach.py]: Please add missing docstrings for all methods.

File: rl_coach/training_worker.py [hunk 9]
```diff
@@ -29,7 +30,7 @@ def data_store_ckpt_save(data_store):
         time.sleep(10)
 
 
-def training_worker(graph_manager, task_parameters):
+def training_worker(graph_manager, task_parameters, multi_node_test):
```
Comment (galleibo-intel) [inline rl_coach/training_worker.py]: Missing parameter docs. Would be nice to add docs across the file. Also, I would change `multi_node_test` to `is_multi_node_test` to make it clear that this is a boolean.

**Comment by timokau** [comment]
Why did this require pinning the kubernetes version to 8.0.1?

**Pull request merged**